### PR TITLE
Unify config for transaction factories

### PR DESCRIPTION
**Pull request by andreibancioiu**
References:
 - https://github.com/multiversx/mx-chain-mainnet-config/tree/master/gasSchedules

### CODE REVIEW

File: multiversx_sdk_core/transaction_factories/transaction_factory_config.py [hunk 1]
```diff
@@ -25,3 +27,12 @@ def __init__(self, chain_id: IChainID) -> None:
...
         self.esdt_contract_address: IAddress = Address.from_bech32("erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u")
+
+        # Configuration for delegation operations
+        self.gas_limit_stake = 5000000
+        self.gas_limit_unstake = 5000000
+        self.gas_limit_unbond = 5000000
+        self.gas_limit_create_delegation_contract = 50000000
+        self.gas_limit_delegation_operations = 1000000
+        self.additional_gas_limit_per_validator_node = 6000000
+        self.additional_gas_for_delegation_operations = 10000000
```
Comment (popenta) [inline multiversx_sdk_core/transaction_factories/transaction_factory_config.py:38]: Perhaps also add underscores to these values as well. Helps with consistency and makes the values more readable.

**Review by popenta — APPROVED** [review]

File: multiversx_sdk_core/transaction_factories/transaction_factory_config.py (hunk 1 above)
Comment (andreibancioiu) [reply multiversx_sdk_core/transaction_factories/transaction_factory_config.py]: Indeed, fixed.

**Review by popenta — APPROVED** [review]

File: multiversx_sdk_core/transaction_factories/delegation_factory.py [hunk 2]
```diff
@@ -90,7 +82,7 @@ def add_nodes(self,
             sender=sender,
             receiver=delegation_contract,
             data_parts=parts,
-            execution_gas_limit=self.config.delegation_ops + num_nodes * self.config.additional_gas_limit_per_node,
+            execution_gas_limit=self.config.gas_limit_delegation_operations + num_nodes * self.config.additional_gas_limit_per_validator_node,
```
Comment (camilbancioiu) [inline multiversx_sdk_core/transaction_factories/delegation_factory.py:85]: Optional:

`self.compute_gas_for_add_nodes(num_nodes)`

File: multiversx_sdk_core/transaction_factories/delegation_factory.py [hunk 3]
```diff
@@ -115,7 +107,7 @@ def remove_nodes(self,
...
```
Comment (camilbancioiu) [inline multiversx_sdk_core/transaction_factories/delegation_factory.py:110]: Optional:

`self.compute_gas_for_remove_nodes(num_nodes)`

File: multiversx_sdk_core/transaction_factories/delegation_factory.py [hunk 4]
```diff
@@ -16,23 +16,15 @@
 
 class IConfig(Protocol):
     chain_id: IChainID
-    min_gas_price: IGasPrice
     min_gas_limit: IGasLimit
     gas_limit_per_byte: IGasLimit
```
Comment (danielailie) [inline multiversx_sdk_core/transaction_factories/delegation_factory.py:20]: why do you initialize this properties, as you have them already initialized on transaction_factory_config?

Comment (andreibancioiu) [reply multiversx_sdk_core/transaction_factories/delegation_factory.py:20]: Good point, will remove them in a separate PR (next one).

Comment (andreibancioiu) [reply multiversx_sdk_core/transaction_factories/delegation_factory.py:20]: Fixed in https://github.com/multiversx/mx-sdk-py-core/pull/29.

File: multiversx_sdk_core/transaction_factories/delegation_factory.py (hunk 2 above)
Comment (andreibancioiu) [reply multiversx_sdk_core/transaction_factories/delegation_factory.py:85]: Fixed in https://github.com/multiversx/mx-sdk-py-core/pull/29.

**Pull request merged**